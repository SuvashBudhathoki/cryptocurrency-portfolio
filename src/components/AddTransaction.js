import React from "react";
import { connect } from "react-redux";
import TransactionForm from "./TransactionForm";
import { addTransaction } from "../actions/transactions";

export class AddTransaction extends React.Component {
  onSubmit = transaction => {
    this.props.addTransaction(transaction);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Transaction</h1>{" "}
          </div>
        </div>
        <div className="content-container">
          <TransactionForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addTransaction: transaction => dispatch(addTransaction(transaction))
});

export default connect(undefined, mapDispatchToProps)(AddTransaction);
