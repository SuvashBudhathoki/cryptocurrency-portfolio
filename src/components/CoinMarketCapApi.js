import React from "react";
import axios from "axios";
import currencyNames from "./CurrencyNames";
import numeral from "numeral";

const apiKey = "19aff428-0470-434b-8807-cf104d68babc";

export default class CoinMarketCap extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      cryptos: []
    };
  }

  componentDidMount() {
    axios
      .get(
        "https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest?CMC_PRO_API_KEY=" +
          apiKey
      )
      .then(res => {
        const cryptos = res.data.data;

        this.setState(() => ({ cryptos }));
      });
  }

  handleSymbol = (crypto, currencyName) => {
    if (crypto.symbol === currencyName.symbol) {
      return (
        <div>
          {crypto.name} {crypto.symbol}
          {numeral(crypto.quote.USD.price / 0.67).format("$0,0.00")}AUD
        </div>
      );
    }
  };

  render() {
    return (
      <div>
        {this.state.cryptos.map((crypto = { id, name, symbol, quote }) =>
          currencyNames.map((currencyName = { symbol }) =>
            this.handleSymbol(crypto, currencyName)
          )
        )}
      </div>
    );
  }
}
