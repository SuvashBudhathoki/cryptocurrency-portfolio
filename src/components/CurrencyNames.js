export default [
  { value: "showAll", label: "Show All Currencies" },
  { value: "bitcoin", label: "Bitcoin", symbol: "BTC" },
  { value: "bitcoinCash", label: "Bitcoin Cash", symbol: "BCH" },
  { value: "ethereumClassic", label: "Ethereum Classic", symbol: "ETC" },
  { value: "ethereum", label: "Ethereum", symbol: "ETH" },
  { value: "litecoin", label: "Litecoin", symbol: "LTC" },
  { value: "ripple", label: "Ripple XRP", symbol: "XRP" },
  { value: "stellar", label: "Stellar", symbol: "XLM" },
  { value: "zCash", label: "Zcash", symbol: "ZEC" }
];
