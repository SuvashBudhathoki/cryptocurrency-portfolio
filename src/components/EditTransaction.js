import React from "react";
import { connect } from "react-redux";
import TransactionForm from "./TransactionForm";
import { editTransaction, removeTransaction } from "../actions/transactions";

export class EditTransaction extends React.Component {
  onSubmit = transaction => {
    this.props.editTransaction(this.props.transaction.id, transaction);
    this.props.history.push("/");
  };
  onRemove = () => {
    this.props.removeTransaction({ id: this.props.transaction.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <TransactionForm
          onSubmit={this.onSubmit}
          transaction={this.props.transaction}
        />
        <button onClick={this.onRemove}>Delete Transaction</button>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editTransaction: (id, transaction) =>
    dispatch(editTransaction(id, transaction)),
  removeTransaction: id => dispatch(removeTransaction(id))
});

const mapStateToProps = (state, props) => ({
  transaction: state.transactions.find(
    transaction => transaction.id === props.match.params.id
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTransaction);
