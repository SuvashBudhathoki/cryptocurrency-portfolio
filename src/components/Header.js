import React from "react";
import { NavLink } from "react-router-dom";

const Header = () => (
  <header>
    <h1>Cryptocurrency</h1>
    <NavLink to="/" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create">Create Transaction</NavLink>
  </header>
);

export default Header;
