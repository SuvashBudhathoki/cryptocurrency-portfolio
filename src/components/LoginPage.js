import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

const LoginPage = ({ startLogin }) => (
  <div class="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Cryptocurrency Portfolio</h1>
      <p>Time to manage your cryptocurrenies</p>
      <button className="button" onClick={startLogin}>
        Login with Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
