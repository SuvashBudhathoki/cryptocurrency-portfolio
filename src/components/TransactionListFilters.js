import React from "react";
import { connect } from "react-redux";
import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import options from "./CurrencyNames";
import "react-dates/lib/css/_datepicker.css";
import Select from "react-select";
import { setTextFilter, sortByAmount, sortByDate } from "../actions/filters";

class TransactionListFilters extends React.Component {
  state = {
    calendarFocused: null,
    selectedOption: "Select your currency from here"
  };

  //Setting up currency name from user

  onValueChange = selectedOption => {
    this.props.dispatch(setTextFilter(selectedOption.label));
    this.setState(() => ({ selectedOption }));
  };

  //Sort By

  onSortChange = e => {
    if (e.target.value === "date") {
      this.props.dispatch(sortByDate());
    } else if (e.target.value === "amount") {
      this.props.dispatch(sortByAmount());
    }
  };

  onFocusChange = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };

  render() {
    return (
      <div>
        <Select
          type="text"
          value={this.state.selectedOption}
          onChange={this.onValueChange}
          options={options}
          placeholder={this.state.selectedOption}
        />
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="date"> Date</option>
          <option value="amount"> Amount</option>
        </select>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    filters: state.filters
  };
};

export default connect(mapStateToProps)(TransactionListFilters);
