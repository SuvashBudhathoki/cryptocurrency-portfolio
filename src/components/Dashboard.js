import React from "react";
import TransactionList from "./TransactionList";
import TransactionTotal from "./TransactionTotal";
import CoinMarketCap from "./CoinMarketCapApi";
import TransactionListFilters from "./TransactionListFilters";

const Dashboard = () => (
  <div>
    <p>
      <TransactionListFilters />
      <TransactionTotal />
      <TransactionList />
      ------CoinMarketCap----
    </p>
  </div>
);

export default Dashboard;
