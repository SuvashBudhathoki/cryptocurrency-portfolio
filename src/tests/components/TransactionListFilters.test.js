import React from "react";
import { shallow } from "enzyme";
import { TransactionListFilters } from "../../components/TransactionListFilters";
import { filters, altFilters } from "../fixtures/fliters";

let setTextFilter, sortByDate, sortByAmount, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByAmount = jest.fn();
  sortByDate = jest.fn();
  wrapper = shallow(
    <TransactionListFilters
      filters={filters}
      setTextFilter={setTextFilter}
      sortByAmount={sortByAmount}
      sortByDate={sortByDate}
    />
  );
});

//TestCase to ensure TransactionListFilter renders correctly

test("should render TransactionListFilter correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

//TestCase to render with data

test("should render TransactionListFilter with alt data correctly", () => {
  wrapper.setProps({
    filters: altFilters
  });
  expect(wrapper).toMatchSnapshot();
});

//TestCase for sorByDate

test("should sort by date", () => {
  const value = "date";
  wrapper.setProps({
    filters: altFilters
  });
  wrapper.find("select").simulate("change", { target: { value } });
  expect(sortByDate).toHaveBeenCalled();
});

//TestCase for sortByAmount

test("should sort by Amount", () => {
  const value = "amount";

  wrapper.find("select").simulate("change", { target: { value } });
  expect(sortByAmount).toHaveBeenCalled();
});
