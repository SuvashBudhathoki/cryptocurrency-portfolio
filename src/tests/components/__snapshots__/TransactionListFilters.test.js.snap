// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TransactionListFilter correctly 1`] = `
<div>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Show All Currencies",
          "value": "showAll",
        },
        Object {
          "label": "Bitcoin",
          "symbol": "BTC",
          "value": "bitcoin",
        },
        Object {
          "label": "Bitcoin Cash",
          "symbol": "BCH",
          "value": "bitcoinCash",
        },
        Object {
          "label": "Ethereum Classic",
          "symbol": "ETC",
          "value": "ethereumClassic",
        },
        Object {
          "label": "Ethereum",
          "symbol": "ETH",
          "value": "ethereum",
        },
        Object {
          "label": "Litecoin",
          "symbol": "LTC",
          "value": "litecoin",
        },
        Object {
          "label": "Ripple XRP",
          "symbol": "XRP",
          "value": "ripple",
        },
        Object {
          "label": "Stellar",
          "symbol": "XLM",
          "value": "stellar",
        },
        Object {
          "label": "Zcash",
          "symbol": "ZEC",
          "value": "zCash",
        },
      ]
    }
    placeholder="Select your currency from here"
    type="text"
    value="Select your currency from here"
  />
  <select
    onChange={[Function]}
    value="date"
  >
    <option
      value="date"
    >
       Date
    </option>
    <option
      value="amount"
    >
       Amount
    </option>
  </select>
</div>
`;

exports[`should render TransactionListFilter with alt data correctly 1`] = `
<div>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Show All Currencies",
          "value": "showAll",
        },
        Object {
          "label": "Bitcoin",
          "symbol": "BTC",
          "value": "bitcoin",
        },
        Object {
          "label": "Bitcoin Cash",
          "symbol": "BCH",
          "value": "bitcoinCash",
        },
        Object {
          "label": "Ethereum Classic",
          "symbol": "ETC",
          "value": "ethereumClassic",
        },
        Object {
          "label": "Ethereum",
          "symbol": "ETH",
          "value": "ethereum",
        },
        Object {
          "label": "Litecoin",
          "symbol": "LTC",
          "value": "litecoin",
        },
        Object {
          "label": "Ripple XRP",
          "symbol": "XRP",
          "value": "ripple",
        },
        Object {
          "label": "Stellar",
          "symbol": "XLM",
          "value": "stellar",
        },
        Object {
          "label": "Zcash",
          "symbol": "ZEC",
          "value": "zCash",
        },
      ]
    }
    placeholder="Select your currency from here"
    type="text"
    value="Select your currency from here"
  />
  <select
    onChange={[Function]}
    value="amount"
  >
    <option
      value="date"
    >
       Date
    </option>
    <option
      value="amount"
    >
       Amount
    </option>
  </select>
</div>
`;
